### PR TITLE
Nukie win conditions

### PR DESCRIPTION
**Pull request by vulppine**
## About the PR 

Adds a number of ways that nuclear operatives and the crew can win or fail when the gamemode is nuclear operatives.

These are:

- **Syndicate Major Victory**
  - The nuclear bomb explodes on the station.
  - The crew escapes, but the nuclear bomb was armed when they escaped.
  - The nuclear bomb was **delivered to Central Command**.
- **Syndicate Minor Victory**
  - The crew escapes, but leaves all nukies alive.
  - The crew escapes, and some nukies are alive, but they don't bring the disk with them.
- **Neutral Victory**
  - The nuclear bomb explodes off-station.
- **Crew Minor Victory**
  - The crew escapes, and some nukies are alive, but they manage to secure the disk.
- **Crew Major Victory**
  - All nuclear operatives die.
  - The crew manages to blow up the nuclear operative outpost.

**Changelog**

:cl:
- tweak: Nanotrasen and the Syndicate have provided guidelines to their crews as to how to consider clashes against nuclear operations as successes or failures.

### CODE REVIEW

**Comment by Just-a-Unity-Dev** [comment]
can you add a "Nukies are antag-banned from playing nukies ever again" when the nuke explodes on the syndi base

**Comment by Just-a-Unity-Dev** [comment]
can you antag-ban the nukies on crew major

**Comment by GoodWheatley** [comment]
Know it’s a bit late to chime in, but I’d like to see the nuke being delivered to CentCom classified as a “Overwhelming Syndicate Victory” and the crew blowing up the nuke planet a “Overwhelming Crew Victory”.

**Comment by vulppine** [comment]
> Know it’s a bit late to chime in, but I’d like to see the nuke being delivered to CentCom classified as a “Overwhelming Syndicate Victory” and the crew blowing up the nuke planet a “Overwhelming Crew Victory”.

this actually was the original intent (Syndicate/Crew humiliation victories) but it's more like a little edge case that's not necessarily the goal, but more a fun little 'oh, you can win that way?' message

File: Content.Server/Nuke/NukeSystem.cs [hunk 1]
```diff
@@ -519,7 +519,10 @@ public void ToggleBomb(EntityUid uid, NukeComponent? component = null)
                 component.IntensitySlope,
                 component.MaxIntensity);
 
-            RaiseLocalEvent(new NukeExplodedEvent());
+            RaiseLocalEvent(new NukeExplodedEvent()
+            {
+                OwningStation = Transform(uid).GridUid,
+            });
```
Comment (metalgearsloth) [inline Content.Server/Nuke/NukeSystem.cs:525]: Transform is already resolved above.

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 2]
```diff
@@ -55,23 +106,209 @@ public override void Initialize()
...
+        {
+            if (ev.OwningStation == _outpostGrid)
+            {
+                _winType = WinType.CrewMajor;
+                _winConditions.Add(WinCondition.NukeExplodedOnNukieOutpost);
+            }
+            else
+            {
+                if (_targetStation != null
+                    && TryComp(_targetStation, out StationDataComponent? data))
```
Comment (metalgearsloth) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:127]: You should just be able to TryComp(_targetStation)

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 3]
```diff
@@ -55,23 +106,209 @@ public override void Initialize()
...
+
+    private void OnRoundEnd()
+    {
+        // If the win condition was set to operative/crew major win, ignore.
+        if (_winType == WinType.OpsMajor || _winType == WinType.CrewMajor)
+        {
+            return;
+        }
+
+        foreach (var nuke in EntityManager.EntityQuery<NukeComponent>())
```
Comment (metalgearsloth) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:198]:
```suggestion
        foreach (var nuke in EntityManager.EntityQuery<NukeComponent>(true))
```

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 4]
```diff
@@ -55,23 +106,209 @@ public override void Initialize()
...
+        {
+            if (nuke.Status != NukeStatus.ARMED)
+            {
+                continue;
+            }
+
+            var nukeTransform = Transform(nuke.Owner);
```
Comment (metalgearsloth) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:205]: Just use EntityQuery<NukeComponent, TransformComponent>

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 5]
```diff
@@ -55,23 +106,209 @@ public override void Initialize()
...
+            _winConditions.Add(WinCondition.AllNukiesAlive);
+            return;
+        }
+
+        _winConditions.Add(WinCondition.SomeNukiesAlive);
+
+        var diskAtCentCom = false;
+        foreach (var comp in EntityManager.EntityQuery<NukeDiskComponent>())
+        {
+            var diskMapId = Transform(comp.Owner).MapID;
```
Comment (metalgearsloth) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:253]: As above i.e. true + EntityQuery<NukeDiskComponent, TransformComponent>

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by vulppine** [comment]
blocked until #10005 is merged

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**